web-src/src/helpers/ResultsParser.js

```diff
@@ -14,9 +14,55 @@ function objectToString(obj) {
   return String(obj).replace(/<\/?[^>]+(>|$)/g, '');
 }
 
+/**
+ * Attempts to parse a potentially incomplete JSON string.
+ *
+ * @param {string} str - The JSON string to parse.
+ * @returns {any} - The parsed JSON object, a best-effort fix if the input is incomplete,
+ *                  or the original string if parsing fails.
+ *
+ * The function works as follows:
+ * 1. Tries to parse the input string using `JSON.parse`.
+ * 2. If parsing fails, it invokes an internal `fixJson` function:
+ *    - Checks the first character to determine if the JSON represents an array or object.
+ *    - Finds the last closing curly brace (`}`) in the string.
+ *    - Truncates the string up to the last closing brace and attempts to close it correctly.
+ *    - Parses the truncated string. If this fails, returns the original incomplete string.
+ *    - If no closing brace is found, returns an empty object `{}` for objects or an empty array `[]` for arrays.
+ */
+const parseResponse = (str) => {
+  const fixJson = (incomplete) => {
+    const firstChar = incomplete.substring(0, 1);
+    const isArray = firstChar === '[';
+    const isObject = firstChar === '{';
+    if (!isArray && !isObject) {
+      return incomplete;
+    } else {
+      const lastClosingCurlyBrace = incomplete.lastIndexOf('}');
```

Comment:
i think you can modify this condition to satisfy array and obj use cases:
```suggestion
     const lastClosingCurlyBrace =  isArray ? incomplete.lastIndexOf(']') : incomplete.lastIndexOf('}')
```

Reply:
Right now, we should build prompts in a way that ensures we get a response as an array of objects. Therefore, an additional check for whether we get an object might be unnecessary. However, it doesn't hurt to have it.